plantswap/components/Results/Results.js

```diff
@@ -3,71 +3,114 @@ import Title from "../Title/Title";
 import styles from "./results.module.scss";
 import Link from "next/link";
 import Text from "../Text/Text";
+import Button from "../Button/Button";
+import Rank from "./Rank";
+import ucFirst from "../../utils/ucFirst";
 // import dataset from "../Switcher/"
-const Results = ({ data }) => {
-
-  if (!data) return <Title title={"h1"}>Geen groene vrienden gevonden</Title>;
-  const getClassByIndex = (index) => {
-    if (index === 0) {
-      return styles.first;
-    } else if (index === 1) {
-      return styles.second;
-    } else if (index === 2) {
-      return styles.third;
-    } else {
-      return '';
-    }
-  };
+const Results = ({ mydata, data }) => {
+  const { flowdata, dictionary } = mydata;
+  if (!data)
+    return <Title title={"h2"}>{dictionary.swapflow.myplant.error}</Title>;
   return (
-    
     <section className={styles.results}>
-      <Title title={"h1"}>Resultaten</Title>
-      <Title className={styles.results__resultlength} title={"h2"}>{data.results.length} groene kanidaten gevonden</Title>
+      <Title title={"h2"}>{dictionary.swapflow.myplant.title}</Title>
+      <Title className={styles.results__count} title={"h3"} modifier={"small"}>
+        {data.results.length} {ucFirst(dictionary.plantsNickName)}
+      </Title>
 
-{
-  data.results.map((result, i) => (
-    <>
-    <Result
-      result={result}
-      key={i}
-      className={`${styles.result} ${getClassByIndex(i)}`}
-    />
-    {i === 2 && <Text className={styles.result__remaining} title={"p"}>Resterende {data.results.length - 3} groene kanidaten.</Text>}
-    </>
-  ))
-  
-}
+      {data.results.map((result, i) => (
+        <Result
+          key={i}
+          flowData={flowdata.flowData}
+          setFlowData={flowdata.setFlowData}
+          dictionary={dictionary}
+          result={result}
+          bestQuess={i <= 2}
+          position={i + 1}
+          totolResults={data.results.length}
+          // className={`${styles.result} ${getClassByIndex(i)}`}
+        />
+      ))}
     </section>
-    
   );
-  
 };
 
-export const Result = ({ result, className }) => {
-  // console.log(className)
+export const Result = ({
+  bestQuess,
+  position,
+  result,
+  dictionary,
+  flowData,
+  setFlowData,
+  totolResults,
+}) => {
+  console.log(position);
+  // const { flowData, setFlowData } = mydata.flowdata;
+  // console.log(mydata.flowdata.setFlowData);
   return (
-    <section className={`${styles.result} ${className}`}>
+    <>
+      <section
+        data-position={position}
+        className={`${styles.result} ${
+          bestQuess && styles[`result--best-quess`]
+        }`}>
+        <Title className={styles.result__name} title={"h4"}>
+          {result.species.commonNames[0]}
+        </Title>
+        <Title className={styles.result__latin} title={"h5"}>
```

Comment:
is het niet beter om dit een P element te maken? Niet alle namen hoeven in header te zijn. vooral als het geen vervolg text heeft